File: Numbers/fibonacci.py
#Anindya Guha

def fibonacci(n):
    """
    A memoized implementation of (a function
    that generates) the fibonacci sequence, upto the nth digit.
    """
    if n==0:
    	return None
    fibarray = [0 for i in range(n)]
    fibarray[0] = 0
    fibarray[1] = 1
    for i in range(2, n):
    	fibarray[i] = fibarray[i-1] + fibarray[i-2]
    [print(ch, end=" ") for ch in fibarray]
    return fibarray

n = input()
fibonacci(int(n))

File: Numbers/fibonacci.php
<?php
    /* This takes a number, n, from STDIN, and generates the fibonacci sequence 
		upto that number.
	*/
	function fibonacci($n)
	{
		$fibarray = array(0, 1);  
		for($i=2; $i<$n; $i++){
			$fibarray[$i] = $fibarray[$i-1] + $fibarray[$i-2];
		}
		foreach ($fibarray as $i){
			echo $i . " ";  
		}
	}
	$number = fgets(STDIN);
	fibonacci($number);
?>

